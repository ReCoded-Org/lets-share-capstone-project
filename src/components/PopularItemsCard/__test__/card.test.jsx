"use strict";

import renderer from "react-test-renderer";

import Card from "../card.jsx";

const item = {
    id: 13,
    title: "Fantastic Frozen Computer",
    location: "Istanbul",
    price: 787,
    description:
        "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
    category: "Furniture",
    itemImage: ["/Photo.png", "/Photo.png", "/Photo.png"],
    images: [
        "https://api.lorem.space/image/furniture?w=640&h=480&r=9644",
        "https://api.lorem.space/image/furniture?w=640&h=480&r=9186",
        "https://api.lorem.space/image/furniture?w=640&h=480&r=6667",
    ],
};

it("renders correctly", () => {
    const tree = renderer.create(<Card item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
});
