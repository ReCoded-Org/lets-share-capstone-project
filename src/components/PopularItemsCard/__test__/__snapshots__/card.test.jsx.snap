// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="group w-[250px] cursor-pointer overflow-hidden rounded-lg bg-[#F3F3F3] text-primary shadow-xl transition duration-300 hover:scale-105 hover:shadow-2xl"
>
  <div
    className="relative aspect-video overflow-hidden rounded-md transition-all "
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "bottom": 0,
          "boxSizing": "border-box",
          "display": "block",
          "height": "initial",
          "left": 0,
          "margin": 0,
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "initial",
        }
      }
    >
      <img
        alt="Sunset in the mountains"
        className="abso h-full w-full"
        data-nimg="fill"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "cover",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="Sunset in the mountains"
          className="abso h-full w-full"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=3840&q=75"
          srcSet="/_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fapi.lorem.space%2Fimage%2Ffurniture%3Fw%3D640%26h%3D480%26r%3D9644&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    className=" flex justify-between py-4 px-3"
  >
    <div
      className=" flex flex-col justify-between"
    >
      <h2
        className="text-brand-primary dark:text-white text-lg font-semibold tracking-normal"
      >
        <span
          className=" dark:from-purple-800 dark:to-purple-900 bg-gradient-to-r from-[#4ade80] to-[#dcfce7] bg-[length:0px_10px] bg-left-bottom bg-no-repeat transition-[background-size] duration-500 hover:bg-[length:100%_3px] group-hover:bg-[length:100%_10px]"
        >
          Fantastic Frozen Computer
        </span>
      </h2>
      <p
        className="text-gray-700 mt-2 text-base "
      >
        The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design
      </p>
    </div>
    <div
      className="flex flex-col items-center justify-between"
    >
      <div
        className="flex"
      >
        <svg
          className="mt-1 h-4 w-4 group-hover:animate-bounce"
          fill="#F07167"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.05 4.05a7 7 0 119.9 9.9L10 18.9l-4.95-4.95a7 7 0 010-9.9zM10 11a2 2 0 100-4 2 2 0 000 4z"
            fillRule="evenodd"
          />
        </svg>
         
        <span
          className="ml-1"
        >
          Adana
        </span>
      </div>
      <span
        className="opacity-0 transition duration-300 group-hover:opacity-100 "
      >
        Furniture
      </span>
    </div>
  </div>
</div>
`;
